psi4/driver/p4util/python_helpers.py

```diff
@@ -185,3 +189,84 @@ def pcm_helper(block):
+    """For PsiAPI mode, forms a basis specification function from *block*
+    and associates it with keyword *key* under handle *name*. Registers
+    the basis spec with Psi4 so that it can be applied again to future
+    molecules. For usage, see mints2, mints9, and cc54 test cases.
+
+    """
+    key = key.upper()
+    name = ('anonymous' + str(random.randint(0, 999))) if name == '' else name
+    cleanbas = basname(name).replace('-', '')  # further remove hyphens so can be function name
+    block = filter_comments(block)
+    command_lines = re.split('\n', block)
+
+    symbol_re = re.compile(r'^\s*assign\s+(?P<symbol>[A-Z]{1,3})\s+(?P<basis>[-+*\(\)\w]+)\s*$', re.IGNORECASE)
+    label_re = re.compile(r'^\s*assign\s+(?P<label>(?P<symbol>[A-Z]{1,3})(?:(_\w+)|(\d+))?)\s+(?P<basis>[-+*\(\)\w]+)\s*$', re.IGNORECASE)
+    all_re = re.compile(r'^\s*assign\s+(?P<basis>[-+*\(\)\w]+)\s*$', re.IGNORECASE)
+    basislabel = re.compile(r'\s*\[\s*([-*\(\)\w]+)\s*\]\s*')
+
+    def anon(mol, role):
+        basstrings = {}
+
+        # Start by looking for assign lines, and remove them
+        leftover_lines = []
+        assignments = False
+        for line in command_lines:
+            if symbol_re.match(line):
+                m = symbol_re.match(line)
+                mol.set_basis_by_symbol(m.group('symbol'), m.group('basis'), role=role)
+                assignments = True
+
+            elif label_re.match(line):
+                m = label_re.match(line)
+                mol.set_basis_by_label(m.group('label'), m.group('basis'), role=role)
+                assignments = True
+
+            elif all_re.match(line):
+                m = all_re.match(line)
+                mol.set_basis_all_atoms(m.group('basis'), role=role)
+                assignments = True
+
+            else:
+                # Ignore blank lines and accumulate remainder
+                if line and not line.isspace():
+                    leftover_lines.append(line.strip())
+
+        # Now look for regular basis set definitions
+        basblock = list(filter(None, basislabel.split('\n'.join(leftover_lines))))
+        if len(basblock) == 1:
+            if not assignments:
+                # case with no [basname] markers where whole block is contents of gbs file
+                mol.set_basis_all_atoms(name, role=role)
+                basstrings[basname(name)] = basblock[0]
+            else:
+                message = ("Conflicting basis set specification: assign lines present but shells have no [basname] label.""")
+                raise TestComparisonError(message)
+        else:
+            # case with specs separated by [basname] markers
+            for idx in range(0, len(basblock), 2):
+                basstrings[basname(basblock[idx])] = basblock[idx + 1]
+
+        return basstrings
+    anon.__name__ = 'basisspec_psi4_yo__' + cleanbas
+    qcdb.libmintsbasisset.basishorde[name.upper()] = anon
+    core.set_global_option(key, name)
```

Comment:
Good point about the docstring. I'm fine with adding a kwarg to suppress setting the option. But at least in Psithon, I regard `df_basis_scf mybas {...}` as just an elaborate `set df_basis_scf mybas` that just happens to need multiline strings. And so the function name and arg order of the PsiAPI `basis_helper()` could use some improvements, but should basically behave the same way, just as `geometry(...)` sets active molecule. So I'm in the default `basis_helper(..., set_option=True`) camp.